jscomp/bsb/bsb_pkg.ml

```diff
@@ -101,7 +101,65 @@ let resolve_bs_package ~cwd (package : t) =
       end;
     x
 
+(* Some package managers will implement "postinstall" caches, that do not
+ * keep their build artifacts in the local node_modules. Similar to
+ * npm_config_prefix, bs_custom_resolution allows these to specify the
+ * exact location of build cache, but on a per-package basis. Implemented as
+ * environment lookup to avoid invasive changes to bsconfig and mandates. *)
+let custom_resolution =
+ match Sys.getenv "bs_custom_resolution" with
+  | exception Not_found  -> false
+  | "true"  -> true
+  | _ -> false
 
+let regex_at = Str.regexp "@"
+let regex_unders = Str.regexp "_+"
+let regex_slash = Str.regexp "\\/"
+let regex_dot = Str.regexp "\\."
+let regex_hyphen = Str.regexp "-"
```

Comment:
we don't use regex unless really have to

Reply:
Having a hard time coming up with a better solution than this:
```ocaml
let package_to_install package =
  Bsb_pkg_types.to_string package
  |> String.split '@'
  |> String.concat ""
  |> String.split '_'
  |> String.concat "__"
  |> String.split '/'
  |> String.concat "__slash__"
  |> String.split '.'
  |> String.concat "__dot__"
  |> String.split '-'
  |> String.concat "_";
```

Reply:
Scratch that, `split_on_char` doesn't exist in 4.02.3

Reply:
@bobzhang I'm not sure how to solve this without regex without doing a ton of work parsing strings and putting them together.

Reply:
@ulrikstrid Ext_string has such functionality